storage/cache/map_subtree_cache.go

```diff
@@ -61,7 +61,7 @@ func populateMapSubtreeNodes(treeID int64, hasher hashers.MapHasher) storage.Pop
 				return nil, nil
 			},
 			func(depth int, index *big.Int, h []byte) error {
-				if depth == 0 {
+				if depth == 0 && len(st.Prefix) > 0 {
```

Comment:
Yep but the log case broke when the tree spanned multiple subtrees so the unit tests were a false sense of security. I think the batch test is probably enough.